plugins/TopPairMuonPlusJetsSelectionFilter.cc

```diff
@@ -301,9 +301,13 @@ bool TopPairMuonPlusJetsSelectionFilter::passesLooseMuonVeto() const {
 			}
 		}
 		return isZEvent == true;
-	} else
-		// Require only one loose muon, which is the signal muons
-		return vetoMuons_.size() < 2;
+	} 	
+	else {
+		if (nonIsolatedMuonSelection1_ || nonIsolatedMuonSelection2_){
+			return vetoElectrons_.size() < 1;
+		}
+		return vetoElectrons_.size() < 2;
```

Comment:
I have realised that this is veto electrons being applied to muon veto... I will change this. I also realise there is still a bug in the muon qcd cr. The loose veto overlaps in isolation with the qcd crs, so this method of rejecting additional leptons will work for some events iso>loose_iso and not for events iso<loose_iso